static/app/components/events/breadcrumbs/breadcrumbsDataSection.tsx

```diff
@@ -0,0 +1,215 @@
+import color from 'color';
+
+import {Button} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import {CompactSelect} from 'sentry/components/compactSelect';
+import DropdownButton from 'sentry/components/dropdownButton';
+import ErrorBoundary from 'sentry/components/errorBoundary';
+import BreadcrumbsTimeline from 'sentry/components/events/breadcrumbs/breadcrumbsTimeline';
+import {
+  applyBreadcrumbSearch,
+  BREADCRUMB_TIME_DISPLAY_LOCALSTORAGE_KEY,
+  BREADCRUMB_TIME_DISPLAY_OPTIONS,
+  BreadcrumbTimeDisplay,
+  getBreadcrumbFilters,
+} from 'sentry/components/events/breadcrumbs/utils';
+import {EventDataSection} from 'sentry/components/events/eventDataSection';
+import {
+  BREADCRUMB_SORT_LOCALSTORAGE_KEY,
+  BREADCRUMB_SORT_OPTIONS,
+  BreadcrumbSort,
+} from 'sentry/components/events/interfaces/breadcrumbs';
+import {PANEL_INITIAL_HEIGHT} from 'sentry/components/events/interfaces/breadcrumbs/breadcrumbs';
+import {getVirtualCrumb} from 'sentry/components/events/interfaces/breadcrumbs/utils';
+import Input from 'sentry/components/input';
+import {IconClock, IconFilter, IconSort} from 'sentry/icons';
+import {t, tn} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {EntryType, type Event} from 'sentry/types';
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+
+interface BreadcrumbsDataSectionProps {
+  event: Event;
+}
+
+export default function BreadcrumbsDataSection({event}: BreadcrumbsDataSectionProps) {
+  const [search, setSearch] = useState('');
+  const [filterSet, setFilterSet] = useState(new Set<string>());
+  const [sort, setSort] = useLocalStorageState<BreadcrumbSort>(
+    BREADCRUMB_SORT_LOCALSTORAGE_KEY,
+    BreadcrumbSort.NEWEST
+  );
+  const [timeDisplay, setTimeDisplay] = useLocalStorageState<BreadcrumbTimeDisplay>(
+    BREADCRUMB_TIME_DISPLAY_LOCALSTORAGE_KEY,
+    BreadcrumbTimeDisplay.RELATIVE
+  );
+
+  const breadcrumbEntryIndex = event.entries.findIndex(
+    entry => entry.type === EntryType.BREADCRUMBS
+  );
+  if (!breadcrumbEntryIndex) {
+    return null;
+  }
+  const breadcrumbs = event.entries[breadcrumbEntryIndex]?.data?.values ?? [];
+  if (breadcrumbs.length <= 0) {
+    return null;
+  }
+
+  const meta = event._meta?.entries?.[breadcrumbEntryIndex]?.data?.values;
+
+  // The virtual crumb is a representation of this event, displayed alongside
+  // the rest of the breadcrumbs for more additional context.
+  const virtualCrumb = getVirtualCrumb(event);
+  let virtualCrumbIndex: number | undefined;
+  const allCrumbs = [...breadcrumbs];
+  if (virtualCrumb) {
+    virtualCrumbIndex = allCrumbs.length;
+    allCrumbs.push(virtualCrumb);
+  }
+
+  const filterOptions = getBreadcrumbFilters(allCrumbs);
+  const filteredCrumbs = allCrumbs.filter(bc =>
+    filterSet.size === 0 ? true : filterSet.has(bc.type)
+  );
+
+  const searchedCrumbs = applyBreadcrumbSearch(search, filteredCrumbs);
```

Comment:
i would imagine you'll need to memo something here at some point. should find some examples of issues with lots of breadcrumbs